stream_processing/streaming_to_datalake.py

```diff
@@ -0,0 +1,158 @@
+MINIO_ENDPOINT = os.getenv("MINIO_ENDPOINT")
+MINIO_ACCESS_KEY = os.getenv("MINIO_ACCESS_KEY")
+MINIO_SECRET_KEY = os.getenv("MINIO_SECRET_KEY")
+BUCKET_NAME = os.getenv("BUCKET_NAME")
+###############################################
+
+
+###############################################
+# PySpark
+###############################################
+def create_spark_session():
+    """
+        Create the Spark Session with suitable configs
+    """
+    from pyspark.sql import SparkSession
+
+    try: 
+        spark = (SparkSession.builder.config("spark.executor.memory", "4g") \
+                        .config(
+                            "spark.jars.packages", "org.apache.spark:spark-sql-kafka-0-10_2.12:3.0.0,org.apache.hadoop:hadoop-aws:2.8.2"
+                        )
+                        .config("spark.hadoop.fs.s3a.access.key", MINIO_ACCESS_KEY)
+                        .config("spark.hadoop.fs.s3a.secret.key", MINIO_SECRET_KEY)
+                        .config("spark.hadoop.fs.s3a.endpoint", MINIO_ENDPOINT)
+                        .config("spark.hadoop.fs.s3a.aws.credentials.provider", "org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider")
+                        .config("spark.hadoop.fs.s3a.path.style.access", "true")
+                        .config("spark.hadoop.fs.s3a.connection.ssl.enabled", "false")
+                        .config("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+                        .config("spark.sql.execution.arrow.pyspark.enabled", "true")
+                        .appName("Streaming Processing Application")
+                        .getOrCreate()
+        )
+        
+        logging.info('Spark session successfully created!')
+
+    except Exception as e:
+        traceback.print_exc(file=sys.stderr)
+        logging.error(f"Couldn't create the spark session due to exception: {e}")
+
+    return spark
+
+
+def create_initial_dataframe(spark_session):
+    """
+        Reads the streaming data and creates the initial dataframe accordingly
+    """
+    try: 
+        df = (spark_session
+            .readStream
+            .format("kafka")
+            .option("kafka.bootstrap.servers", "localhost:9092")
+            .option("subscribe", "device.iot.taxi_nyc_time_series")
+            # .option("startingOffsets", "earliest")
+            .option("failOnDataLoss", "false")
+            .load())
+        logging.info("Initial dataframe created successfully!")
+    except Exception as e:
+        logging.warning(f"Initial dataframe could not be created due to exception: {e}")
+
+    return df
+
+
+def create_final_dataframe(df, spark_session):
+    """
+        Modifies the initial dataframe, and creates the final dataframe
+    """
+    from pyspark.sql.types import IntegerType, StringType, StructType, StructField, TimestampNTZType, DoubleType, LongType
+    from pyspark.sql.functions import col, from_json, udf
+
+    payload_after_schema = StructType([
```

Comment:
Try to put these long lines into a config file!

Reply:
I don't quite understand this comment, can you guide me?

Reply:
Create a mapping file based on this code, then import it into this Python code using a JSON file. Each key represents a field name, and the values indicate the type and true/false.

Reply:
Fixed!